integration/container/exec_test.go

```diff
@@ -117,3 +117,53 @@ func TestExec(t *testing.T) {
 	assert.Assert(t, is.Contains(out, "PWD=/tmp"), "exec command not running in expected /tmp working directory")
 	assert.Assert(t, is.Contains(out, "FOO=BAR"), "exec command not running with expected environment variable FOO")
 }
+
+func TestExecKill(t *testing.T) {
+	skip.If(t, versions.LessThan(testEnv.DaemonAPIVersion(), "1.40"), "broken in earlier versions")
+	skip.If(t, testEnv.OSType == "windows", "FIXME. Probably needs to wait for container to be in running state.")
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := testEnv.APIClient()
+
+	cID := container.Run(t, ctx, client, container.WithTty(true), container.WithWorkingDir("/root"))
+
+	// Assume this test ends within 10 seconds
+	id, err := client.ContainerExecCreate(ctx, cID,
+		types.ExecConfig{
+			Detach: true,
+			Cmd:    strslice.StrSlice([]string{"sh", "-c", "for i in {1..10}; do sleep 1; done"}),
+		},
+	)
+	assert.NilError(t, err)
+
+	resp, err := client.ContainerExecAttach(ctx, id.ID, types.ExecStartCheck{})
+	assert.NilError(t, err)
+	defer resp.Close()
+
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	select {
+	case <-time.After(5 * time.Second):
```

Comment:
how can this happen

Reply:
The state of a process is checked periodically just below this line. I'd like to make sure if it's running.

Reply:
IIUC `case <-ticker.C:` is always executed after 1sec? Then `case <-time.After(5 * time.Second):` won't be executed?

Reply:
Yes, that's true. However, I put the safety guard just in case if it doesn't finish in a few seconds. I'll remove it and put sleep of 1 second if it's better.

Reply:
Please refer to https://github.com/moby/moby/blob/c8e398b96baeb70a572af83264ce74e29a9ce597/integration/container/exec_test.go#L51-L83

Reply:
I think the code you mentioned is to wait for closed connection which can block until disconnection. The test I wrote uses `detach` option in `ContainerExecCreate`, and `ContainerExecInspect` may return not found error or a response with `Running=false` without blocking. Could you tell me how to wait for an active exec instance in more detail?